src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/IStartupProjectHelper.cs

```diff
@@ -1,14 +1,16 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 
+using System.Collections.Generic;
+
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
 {
     /// <summary>
-    /// Provides a mechanism to get an export from a DotNet project which is the only startup project. The capabilityMatch is  
+    /// Provides a mechanism to get an export from a DotNet project which is the only startup project. The capabilityMatch is
     /// used to refine the projects that are considered
     /// </summary>
     internal interface IStartupProjectHelper
     {
-        T GetExportFromSingleDotNetStartupProject<T>(string capabilityMatch) where T : class;
+        List<T> GetExportFromDotNetStartupProjects<T>(string capabilityMatch) where T : class;
```

Comment:
Does this have to return `List<T>`? Can you use `ImmutableArray<T>` or `IReadOnlyList<T>` instead?

Reply:
Either of those options should work. Which one should I change it to?

Reply:
The solution has the following usages:

- 34 `IReadOnlyList<T>`
- 15 `IImmutableList<T>`
- 60 `ImmutableList<T>`
- 168 `ImmutableArray<T>`

Looks like `ImmutableArray<T>` would be the more common choice, especially if the return value is not being manipulated (as the array variant copies internal storage on modification, whereas the list variant shares internal structure).

Reply:
I've switched to `ImmutableArray<T>`.